nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java

```diff
@@ -107,7 +108,8 @@ public boolean onOverrideMayBeNullExpr(
     if (expr.getKind() == Tree.Kind.METHOD_INVOCATION) {
       OptimizedLibraryModels optLibraryModels = getOptLibraryModels(state.context);
       Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) ASTHelpers.getSymbol(expr);
-      if (!NullabilityUtil.isUnannotated(methodSymbol, this.config)) {
+      if (!NullabilityUtil.isUnannotated(
+          methodSymbol, this.config, NullMarkedCache.instance(state.context))) {
```

Comment:
Again, these handler calls are very frequent. Is `context.get(NULL_MARKED_CACHE_KEY);` literally just a map lookup or something more heavyweight?

Reply:
Edit: Never mind. This looks to be a map key lookup to me, as long as the IntelliJ decompiler can be believed and we are using the EP javac9 (probably similar for JDK 11/JDK 17?)

Reply:
There is code in Error Prone that claims these lookups can be expensive:

https://github.com/google/error-prone/blob/804cdbdf40ffb8ed093d7be49567db33af3d38d6/check_api/src/main/java/com/google/errorprone/VisitorState.java#L704-L709

I guess  a field read is cheaper than a map lookup.  I'll play it safe and just add instance fields in all classes using `NullMarkedCache` to hold the instance once it has been looked up.

Reply:
Ok. Makes sense to me. My feeling is `instance(...)` speed depends on JDK version/implementation and, in any case, can't be any faster than null check + instance field dereference.